tfhe/src/integer/server_key/radix_parallel/ilog2.rs

```diff
@@ -0,0 +1,734 @@
+        };
+
+        let input_values = [-modulus, 0i64, modulus - 1]
+            .into_iter()
+            .chain((0..NB_TESTS_SMALLER).map(|_| rng.gen::<i64>() % modulus))
+            .collect::<Vec<_>>();
+
+        for clear in input_values {
+            let ctxt = cks.encrypt_signed(clear);
+
+            let ct_res = sks_method(&sks, &ctxt);
+            let tmp = sks_method(&sks, &ctxt);
+            assert!(ct_res.block_carries_are_empty());
+            assert_eq!(ct_res, tmp);
+
+            let decrypted_result: u32 = cks.decrypt(&ct_res);
+            let expected_result = compute_expected_clear(clear);
+            assert_eq!(
+                decrypted_result, expected_result,
+                "Invalid result for {method_name}, for {clear}.{method_name}() \
+                expected {expected_result}, got {decrypted_result}"
+            );
+
+            for _ in 0..NB_TESTS_SMALLER {
+                // Add non-zero scalar to have non-clean ciphertexts
+                let clear_2 = random_non_zero_value(&mut rng, modulus);
+
+                let ctxt = sks.unchecked_scalar_add(&ctxt, clear_2);
+
+                let clear = signed_add_under_modulus(clear, clear_2, modulus);
+
+                let d0: i64 = cks.decrypt_signed(&ctxt);
+                assert_eq!(d0, clear, "Failed sanity decryption check");
+
+                let ct_res = sks_method(&sks, &ctxt);
+                assert!(ct_res.block_carries_are_empty());
+
+                let expected_result = compute_expected_clear(clear);
+
+                let decrypted_result: u32 = cks.decrypt(&ct_res);
+                assert_eq!(
+                    decrypted_result, expected_result,
+                    "Invalid result for {method_name}, for {clear}.{method_name}() \
+                    expected {expected_result}, got {decrypted_result}"
+                );
+            }
+        }
+
+        let input_values = [-modulus, 0i64, modulus - 1]
+            .into_iter()
+            .chain((0..NB_TESTS_SMALLER).map(|_| rng.gen::<i64>() % modulus));
+
+        for clear in input_values {
+            let ctxt = sks.create_trivial_radix(clear, NB_CTXT);
+
+            let ct_res = sks_method(&sks, &ctxt);
+            assert!(ct_res.block_carries_are_empty());
+
+            let decrypted_result: u32 = cks.decrypt(&ct_res);
+            let expected_result = compute_expected_clear(clear);
+            assert_eq!(
+                decrypted_result, expected_result,
+                "Invalid result for {method_name}, for {clear}.{method_name}() \
+                expected {expected_result}, got {decrypted_result}"
+            );
+        }
+    }
+
+    pub(crate) fn default_trailing_zeros_test<P>(param: P)
+    where
+        P: Into<PBSParameters>,
+    {
+        default_test_count_consecutive_bits(
+            Direction::Trailing,
+            BitValue::Zero,
+            param,
+            ServerKey::trailing_zeros,
+        );
+    }
```

Comment:
just a bit confused with the two ways the tests are defined between unsigned and signed, I think I remember there being a reason but I don't recall

Reply:
Because GPU does not yet have started to have signed support

Reply:
does it prevent us from having the executor for signed for the CPU ?

Reply:
hum no, but I think its more or less planned to be done as GPU starts signed support, i.e, they add signed_add, we/they change the signed_add test to have this executor thing

Reply:
ok